File: 3838-weighted-word-mapping/3838-weighted-word-mapping.go
func mapWordWeights(words []string, weights []int) string {
    ans := []byte{}
    for _, word := range words {
        weight := 0
        for i := 0; i < len(word); i++ {
            weight += weights[word[i] - 'a']
        }

        ans = append(ans, byte('z' - (weight % 26)))
    }

    return string(ans)
}

// array, string
// time: O(nk)
// space: O(n)

File: 3838-weighted-word-mapping/3838-weighted-word-mapping.py
class Solution:
    def mapWordWeights(self, words: List[str], weights: List[int]) -> str:
        ans = []
        for word in words:
            weight = 0
            for i in range(len(word)):
                weight += weights[ord(word[i]) - ord('a')]
        
            ans.append(chr(ord('z') - (weight % 26)))
        
        return ''.join(ans)

# array, string
# time: O(nk)
# space: O(n)